pandas/core/frame.py

```diff
@@ -6426,19 +6431,68 @@ def sort_values(  # type: ignore[override]
         else:
             return result.__finalize__(self, method="sort_values")
 
-    @deprecate_nonkeyword_arguments(version=None, allowed_args=["self"])
+    @overload
     def sort_index(
         self,
+        *,
+        axis: Axis = ...,
+        level: Level | None = ...,
+        ascending: bool | int | Sequence[bool | int] = ...,
```

Comment:
Not clear why you are including `int` and `Sequence[int]` for `ascending`.  It's documented as `bool` or "list of `bool`s"

Reply:
I will look into it - these are the existing annotations for it, I didn't change them.

Reply:
ints are used in this test:
https://github.com/pandas-dev/pandas/blob/0be6fd3af16230365c45d006f9e00ffd0cbfd1d2/pandas/tests/frame/methods/test_sort_index.py#L270

but it is honestly not clear to me why ints are accepted

Reply:
Should `int` and `Sequence[int]` be deprecated? @jreback @simonjayhawkins 

The only use I could imagine might be for large MultiIndices. I didn't test the following: let's say you have ten levels - it is easier to just sort by level `[9]` compared to `[False]*8 + [True]`.

Reply:
hmm, I think it was discussed recently that we could omit annotations for deprecated behavior. This makes total sense to me since the end user can easily add a `type: ignore` if they are not ready to alter their code and these type annotations would discourage the use of deprecated behavior. I think we can extend that to undocumented behavior, whether planning to deprecate or not.

Reply:
> hmm, I think it was discussed recently that we could omit annotations for deprecated behavior. This makes total sense to me since the end user can easily add a `type: ignore` if they are not ready to alter their code and these type annotations would discourage the use of deprecated behavior. I think we can extend that to undocumented behavior, whether planning to deprecate or not.

So does that mean we should change the test referred to here https://github.com/pandas-dev/pandas/pull/46300#discussion_r825323365 so that we use  `True/False` rather than `1/0` in the tests, and we just use `bool` and not `int` in this PR in the type declarations?

Reply:
> So does that mean we should change the test referred to here [#46300 (comment)](https://github.com/pandas-dev/pandas/pull/46300#discussion_r825323365) so that we use `True/False` rather than `1/0` in the tests

No. if it is tested behavior, we should not change that, but IMO ok that the types reflect the documented API to discourage the use of undocumented behavior, whether tested or not.

> and we just use `bool` and not `int` in this PR in the type declarations?

sgtm

Reply:
Thanks, I removed `int` and `Sequence[int]` from the annotations.